integration_tests/logical_binop1.py

```diff
@@ -0,0 +1,50 @@
+from ltypes import i32
+
+def test_issue_1487_1():
+    a : i32
+    b : i32
+    x : i32
+    y : i32
+    or_op1 : i32
+    or_op2 : i32
+    a = 1
+    b = 2
+    x = 0
+    y = 4
+    or_op1 = a or b
+    or_op2 = x or y
```

Comment:
I think we should explicitly be casting the operands to`bool()` (when are not already of `bool` type). For example:

```python
res: bool
res = bool(a) or bool(b)
```

For `or_op1 = a or b`, since `or_op1` is of type `i32`, I think it should be
```console
or_op1 = i32(bool(a) or bool(b))
```

Reply:
It seems the result of `x or y` is an integer, I think we are fine in this case.

Reply:
>It seems the result of x or y is an integer, I think we are fine in this case.

It looks like `or` here is not a logical operation, but instead a selective operation. I think we should have a separate node for this `selective` `or` operation.

Reply:
Maybe we can name it as `SelectiveOr` which takes two arguments both of any type (but equal types). The return value of this `SelectiveOr` is the same as that of the type of one of the operands.